wordpress-mobile/WordPress-Android
# [Gutenberg] Faster editor start

Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/672

Currently, the editor has 2 stages during its startup of Gutenberg mobile:
a. The first is some preprocessing happening before the `EditPostActivity` becomes visible
b. The ReactNative app boot itself

This PR addresses only the first stage and introduces a few performance optimizations:

1. On the Aztec code side, uses https://github.com/wordpress-mobile/AztecEditor-Android/pull/790 which adds a simpler and faster html-parsing method, removing functionality that is not needed when pre-processing a post
2. In the wpandroid code side, it introduces a 1-deep cache to hold parsed html-to-spans text so we avoid parsing it too many times while pre-processing.

As an extra touch that can help with UX, this PR adds a small progress indicator, visible while the Gutenberg RN app is initializing, parses and renders the post for the first time.

Sibling PRs:

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/705
* https://github.com/wordpress-mobile/AztecEditor-Android/pull/790

To test:
1. Load a long post (like the ones described in https://github.com/wordpress-mobile/gutenberg-mobile/issues/672) and verify that everything works as expected and the there is no ANR while opening the post.
2. Notice the circular progress indicator spinning while the editor is loading
3. Notice the progress indicator disappearing once the post gets rendered

Update release notes:
